### ISSUE TITLE
Implement cluster preservation metrics (Category C)

### ISSUE THREAD

**Issue by Trecek** [TalonT-Org/spectral-init#197]
## Summary

Implement the 3 cluster preservation metrics (Category C) that measure how well
UMAP embeddings preserve cell-type cluster structure. These compare clustering
derived from the embedding against the reference cell-type annotations.

## Context

- Plan reference: `docs/merfish-umap-benchmark-plan.md` Batch 4, Issue 4.2
- Metric definitions: `docs/metrics/cluster-preservation.md`
- Category B spatial metrics implemented in #195
- Cell-type labels (`cluster_alias`) are available in `merfish_10k_labels.npz`

### Module: `tests/visual_eval/cluster_metrics.py`

Pure numpy/scipy/sklearn module — no scanpy or anndata dependency.

#### Metric 1: Adjusted Rand Index (ARI)

```python
def cluster_ari(
    embedding: np.ndarray,       # (n, 2) UMAP coords
    reference_labels: np.ndarray, # (n,) ground-truth cell-type IDs
    n_clusters: int | None = None,
) -> float:
```
Apply KMeans clustering to the embedding (using the number of unique reference
labels as `n_clusters` if not specified), then compute ARI between the KMeans
assignments and the reference labels. Range [-0.5, 1.0], 1.0 = perfect agreement.

#### Metric 2: Normalized Mutual Information (NMI)

```python
def cluster_nmi(
    embedding: np.ndarray,
    reference_labels: np.ndarray,
    n_clusters: int | None = None,
) -> float:
```
Same clustering approach as ARI, compute NMI between KMeans assignments and
reference labels. Range [0, 1], 1.0 = perfect agreement.

#### Metric 3: Cell-Type Purity

```python
def celltype_purity(
    embedding: np.ndarray,
    reference_labels: np.ndarray,
    n_clusters: int | None = None,
) -> float:
```
For each KMeans cluster in the embedding, compute the fraction of cells
belonging to the dominant reference label. Return the weighted mean purity
across all clusters. Range [0, 1], 1.0 = each cluster contains only one
cell type.

#### Convenience Function

```python
def compute_cluster_metrics(
    embedding: np.ndarray,
    reference_labels: np.ndarray,
    n_clusters: int | None = None,
    seed: int = 42,
) -> dict:
```
Runs all 3 metrics with a single KMeans fit (shared clustering for consistency)
and returns a dict with keys: `ari`, `nmi`, `celltype_purity`.

## Dependencies

- `numpy`
- `sklearn.cluster.KMeans`
- `sklearn.metrics.adjusted_rand_score`, `normalized_mutual_info_score`

## Testing

- Smoke test with synthetic blobs: perfect label alignment → ARI ≈ 1.0,
  NMI ≈ 1.0, purity ≈ 1.0
- Random label permutation → ARI ≈ 0.0
### API — Function Signatures

- **REQ-API-001:** Each metric function must accept numpy arrays as input (no AnnData, no scanpy objects).
- **REQ-API-002:** Each metric function must return a single float value.
- **REQ-API-003:** A `compute_cluster_metrics()` convenience function must exist that runs all 3 metrics and returns a dict.
- **REQ-API-004:** The module must have zero dependency on scanpy, anndata, or polars.
- **REQ-API-005:** The convenience function must perform KMeans only once and reuse the cluster assignments for all 3 metrics.

### ARI — Adjusted Rand Index

- **REQ-ARI-001:** ARI must be computed via `sklearn.metrics.adjusted_rand_score` comparing KMeans assignments to reference labels.
- **REQ-ARI-002:** KMeans must use the number of unique reference labels as `n_clusters` when not explicitly provided.
- **REQ-ARI-003:** KMeans must use `random_state=42` for reproducibility.

### NMI — Normalized Mutual Information

- **REQ-NMI-001:** NMI must be computed via `sklearn.metrics.normalized_mutual_info_score` comparing KMeans assignments to reference labels.

### PURITY — Cell-Type Purity

- **REQ-PURITY-001:** Purity must compute the fraction of cells belonging to the dominant reference label within each KMeans cluster.
- **REQ-PURITY-002:** The overall purity must be the cluster-size-weighted mean across all KMeans clusters.

### TEST — Smoke Tests

- **REQ-TEST-001:** A smoke test must verify that perfectly aligned labels produce ARI ≈ 1.0, NMI ≈ 1.0, purity ≈ 1.0.
- **REQ-TEST-002:** A smoke test must verify that randomly permuted labels produce ARI ≈ 0.0.
